### ISSUE TITLE
Grafana Dashboard for Gateway Metrics Bounty [$500]

### ISSUE THREAD

**Issue by JJassonn69** [JJassonn69/bounties#2]
# Overview

To enhance the observability and performance monitoring of our [ai-subnet](https://docs.livepeer.org/ai/pipelines/overview#models-on-the-ai-subnet/), we aim to implement a Grafana Dashboard for Gateway Metrics. This dashboard will provide critical insights into the AI-job broadcasting operations performed by the Gateway, which is responsible for relaying AI-job requests to orchestrators across the network.

The implementation of this dashboard is crucial for monitoring and optimizing various aspects of the AI-job requests and orchestrator performance, ensuring that the ai-subnet operates efficiently and effectively.

---

## Required Skillset

- Familiarity with [Grafana](https://grafana.com/).
- Strong [Go](https://go.dev/) programming skills.
- Experience with [Prometheus](https://prometheus.io/) for metrics collection.

---

## Bounty Requirements

1. **Implementation:** Develop a Grafana Dashboard thats easy to import and visualize for Gateway operators in the network.
2. **Functionality:** The dashboard should include the following metrics:
    - **ai_models_requested:** Number of requests per model per pipeline. This metric is already available on the [Cloud Gateway](https://monitor.livepeer.cloud/grafana/d/bdpyx1qkiu6tcc/ai-gateway-metrics?orgId=1&refresh=5m/).
    - **ai_request_latency_score:** Latency score per pipeline to assess orchestrator performance. This metric should indicate the average time taken for a request (e.g., a 1024x1024 image with 25 time steps).
    - **ai_request_price:** The price per unit charged by orchestrators for processing jobs.
    - **ai_request_errors:** The number of errors per pipeline, providing insight into the reliability of the pipelines.
    - **Ticket_value_sent:** The value of the AI tickets sent to the Orchestrators.
    - **Tickets_send:** The total number of tickets sent to the Orchestrators.

These metrics are introduced to the `ai-video` branch and will provide comprehensive insights into the public gateway requests on the AI network, enabling better optimization and resource allocation.

---

## Scope Exclusions

- This bounty does **NOT** cover the addition of new metrics such as `livepeer_current_ai_sessions_total`. These additional metrics can be considered in a future bounty.

---

## Implementation Tips

To understand how to work with Gateway metrics, you can refer to a recent pull request that deals with related functionality:

 [Pull Request #3087](https://github.com/livepeer/go-livepeer/pull/3087/)

Additionally, make sure to:

- **Leverage Existing Metrics:** Review existing metrics related to AI requests on the Gateway, which are already implemented on the [Cloud Gateway](https://monitor.livepeer.cloud/grafana/d/bdpyx1qkiu6tcc/ai-gateway-metrics?orgId=1&refresh=5m/). This will provide an initial template for a starting point.
- **Utilize Documentation:** Check out  [livepeer forum](https://forum.livepeer.org/t/guide-transcoder-monitoring-with-prometheus-grafana/1225/) for basic instructions on how to get started.

---

## How to Apply

1. **Express Your Interest:** Comment on this issue to indicate your interest and explain why you're the ideal candidate for the task.
2. **Wait for Review:** Our team will review expressions of interest and select the best candidate.
3. **Get Assigned:** If selected, we'll assign the GitHub issue to you.
4. **Start Working:** Dive into your task! If you need assistance or guidance, comment on the issue or join the discussions in the `#developer-lounge` channel on our [Discord server](https://discord.gg/livepeer).
5. **Submit Your Work:** Create a pull request in the relevant repository and request a review.
6. **Notify Us:** Comment on this GitHub issue when your pull request is ready for review.
7. **Receive Your Bounty:** We'll arrange the bounty payment once your pull request is approved.
8. **Gain Recognition:** Your valuable contributions will be showcased in our project's [changelog](https://livepeer-ai.productlane.com/changelog).

Thank you for your interest in contributing to our project! 💛

> **⚠️ Warning**
> Please wait for the issue to be assigned to you before starting work. To prevent duplication of effort, submissions for unassigned issues will not be accepted.

**Comment by rickstaa** [comment]
@JJassonn69 thanks for creating this bounty!

>To enhance the observability and performance monitoring of our [ai-subnet](https://docs.livepeer.org/ai/pipelines/overview#models-on-the-ai-subnet/), we aim to implement a Grafana Dashboard for Gateway Metrics. This dashboard will provide critical insights into the AI-job broadcasting operations performed by the Gateway, which is responsible for relaying AI-job requests to orchestrators across the network.
>
>The implementation of this dashboard is crucial for monitoring and optimizing various aspects of the AI-job requests and orchestrator performance, ensuring that the ai-subnet operates efficiently and effectively.

Great description! However, adding some emojis could make it more engaging. Also, it would be cool to mention that [Livepeer Cloud](https://www.livepeer.cloud/) is providing a free-to-use gateway. The metrics implemented in [this pull request](https://github.com/livepeer/go-livepeer/pull/3087) are already being collected by the public gateway on [Grafana](https://monitor.livepeer.cloud/grafana/d/bdpyx1qkiu6tcc/ai-gateway-metrics?orgId=1&refresh=5m), and the dashboard is visible to the entire community on that page 🚀. Additionally, this dashboard provides Orchestrators with an approximate view of the full network traffic, as it is the largest gateway. Also, mention that Livepeer Cloud is working on a more comprehensive system-wide metrics portal [here](https://explorer.livepeer.org/treasury/69112973991711207069799657820129915730234258793790128205157315299386501373337).

> - Strong [Go](https://go.dev/) programming skills.

I don't think go skills are needed for this grafana bounty.
 
> **⚠️ Warning**
> Please wait for the issue to be assigned to you before starting work. To prevent duplication of effort, submissions for unassigned issues will not be accepted.

This can be replaced by the following syntax:

```md
> [!WARNING]
> ...

Github now supports experimental admonitions (See https://github.com/orgs/community/discussions/16925.

**Comment by JJassonn69** [comment]
I will modify these and do a update. Thank you for the feedback.

**Comment by JJassonn69** [comment]
# Overview

To enhance observability and performance monitoring of the AI Subnet, we are implementing a Grafana Dashboard for Gateway Metrics. This dashboard offers critical insights into the AI-job broadcasting operations performed by the Gateway, which relays AI-job requests to orchestrators across the network. Livepeer Cloud currently provides a free-to-use gateway, enabling users to test the network's capabilities. Metrics from this [pull request](https://github.com/livepeer/go-livepeer/pull/3087/) are already being collected by the public gateway and displayed on a [Grafana dashboard](https://monitor.livepeer.cloud/grafana/d/bdpyx1qkiu6tcc/ai-gateway-metrics?orgId=1&refresh=5m), accessible to the entire community. 

This dashboard also provides orchestrators with an approximate view of the full network traffic, given its status as the largest gateway. Additionally, Livepeer Cloud is developing a more comprehensive system-wide metrics portal, which can be explored further [here](https://explorer.livepeer.org/treasury/69112973991711207069799657820129915730234258793790128205157315299386501373337/). 🚀

---

## Required Skillset

- Familiarity with [Grafana](https://grafana.com/).
- Experience with [Prometheus](https://prometheus.io/) for metrics collection.

---

## Bounty Requirements

1. **Implementation:** Develop a Grafana Dashboard thats easy to import and visualize for Gateway operators in the network.
2. **Functionality:** The dashboard should include the following metrics:
    - **ai_models_requested:** Number of requests per model per pipeline. This metric is already available on the [Cloud Gateway](https://monitor.livepeer.cloud/grafana/d/bdpyx1qkiu6tcc/ai-gateway-metrics?orgId=1&refresh=5m/).
    - **ai_request_latency_score:** Latency score per pipeline to assess orchestrator performance. This metric should indicate the average time taken for a request (e.g., a 1024x1024 image with 25 time steps).
    - **ai_request_price:** The price per unit charged by orchestrators for processing jobs.
    - **ai_request_errors:** The number of errors per pipeline, providing insight into the reliability of the pipelines.
    - **Ticket_value_sent:** The value of the AI tickets sent to the Orchestrators.
    - **Tickets_send:** The total number of tickets sent to the Orchestrators.

These metrics are introduced to the `ai-video` branch and will provide comprehensive insights into the public gateway requests on the AI network, enabling better optimization and resource allocation.

---

## Scope Exclusions

- This bounty does **NOT** cover the addition of new metrics such as `livepeer_current_ai_sessions_total`. These additional metrics can be considered in a future bounty.

---

## Implementation Tips

To understand how to work with Gateway metrics, you can refer to a recent pull request that deals with related functionality:

 [Pull Request #3087](https://github.com/livepeer/go-livepeer/pull/3087/)

Additionally, make sure to:

- **Leverage Existing Metrics:** Review existing metrics related to AI requests on the Gateway, which are already implemented on the [Cloud Gateway](https://monitor.livepeer.cloud/grafana/d/bdpyx1qkiu6tcc/ai-gateway-metrics?orgId=1&refresh=5m/). This will provide an initial template for a starting point.
- **Utilize Documentation:** Check out  [livepeer forum](https://forum.livepeer.org/t/guide-transcoder-monitoring-with-prometheus-grafana/1225/) for basic instructions on how to get started.

---

## How to Apply

1. **Express Your Interest:** Comment on this issue to indicate your interest and explain why you're the ideal candidate for the task.
2. **Wait for Review:** Our team will review expressions of interest and select the best candidate.
3. **Get Assigned:** If selected, we'll assign the GitHub issue to you.
4. **Start Working:** Dive into your task! If you need assistance or guidance, comment on the issue or join the discussions in the `#developer-lounge` channel on our [Discord server](https://discord.gg/livepeer).
5. **Submit Your Work:** Create a pull request in the relevant repository and request a review.
6. **Notify Us:** Comment on this GitHub issue when your pull request is ready for review.
7. **Receive Your Bounty:** We'll arrange the bounty payment once your pull request is approved.
8. **Gain Recognition:** Your valuable contributions will be showcased in our project's [changelog](https://livepeer-ai.productlane.com/changelog).

Thank you for your interest in contributing to our project! 💛

> [!WARNING]
> Please wait for the issue to be assigned to you before starting work. To prevent duplication of effort, submissions for unassigned issues will not be accepted.

**Comment by JJassonn69** [comment]
Updated and posted a new issue for final review.
